src/Symfony/Component/HttpKernel/DataCollector/ExceptionDataCollector.php

```diff
@@ -22,16 +23,30 @@
  */
 class ExceptionDataCollector extends DataCollector
 {
+    private $flattener;
+
+    public function __construct(ExceptionFlattener $flattener = null)
+    {
+        $this->flattener = $flattener;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function collect(Request $request, Response $response, \Exception $exception = null)
     {
-        if (null !== $exception) {
-            $this->data = array(
-                'exception' => FlattenException::create($exception),
-            );
+        if (null === $exception) {
+            return;
         }
+
+        $flattenException = $request->attributes->get('exception');
```

Comment:
I think that this is correct. It's optimization, because it is used once created exception:
- https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/HttpKernel/EventListener/ExceptionListener.php#L108
- https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/TwigBundle/Controller/PreviewErrorController.php#L48